source/postprocess/matrix_statistics.cc

```diff
@@ -45,12 +46,13 @@ namespace
            << " MB." << std::endl;
 
     // output number of nonzero elements in matrix. Do so with 1000s separator
-    // since they are frequently large; this is done by using the empty
-    // string locale, but creating std::locale with an empty string causes problems
-    // on some platforms, so catch the exception and ignore
+    // since they are frequently large; this was previously done by using the empty
+    // string locale, but creating std::locale with an empty string caused problems
+    // on some platforms, so the functionaltity yo catch the exception and ignore
+    // is kept here, even though explicitly setting a facet should always work.
     try
       {
-        output.imbue(std::locale(""));
+        output.imbue(std::locale(std::locale(), new aspect::Utilities::ThousandSep));
```

Comment:
Hm, thanks for checking. I double checked in the standard, which is also quiet about who takes over ownership of the facet. Apparently, the locale does. So using `new` then appears correct.